hug/types.py

```diff
@@ -344,11 +343,99 @@ def __doc__(self):
     def __call__(self, value):
         return self.convert(value)[:self.limit]
 
+class Chain(Type):
+    """type for chaining multiple types together"""
+    __slots__ = ('types')
+
+    def __init__(self, *types):
+        self.types = types
+
+    def __call__(self, value):
+        for type_function in self.types:
+            value = type_function(value)
+        return value
+
+class Nullable(Chain):
+    """A Chain types that Allows None values"""
+    __slots__ = ('types')
+
+    def __init__(self, *types):
+        self.types = types
+
+    def __call__(self, value):
+        if value == None:
+            return None
+        else:
+            return super(Nullable, self).__call__(value)
+
+class TypedProperty(object):
+    """class for building property objects for schema objects"""
+    __slots__ = ('name', 'type_func')
+    def __init__(self, name, type_func):
+        self.name = "_" + name
+        self.type_func = type_func
+
+    def __get__(self, instance, cls):
+        return getattr(instance, self.name, None)
+
+    def __set__(self, instance, value):
+        setattr(instance,self.name,self.type_func(value))
+
+    def __delete__(self,instance):
+        raise AttributeError("Can't delete attribute")
+
```

Comment:
Very small nitpick, going forward Pep8 enforces 2 empty lines between top level functions / classes. I've adjusted in a local branch and will merge in soon